pkg/net/host.go

```diff
@@ -0,0 +1,149 @@
+/*
+ * Minio Cloud Storage, (C) 2017 Minio, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net
+
+import (
+	"encoding/json"
+	"errors"
+	"net"
+	"regexp"
+	"strings"
+)
+
+var hostLabelRegexp = regexp.MustCompile("^[a-zA-Z0-9]([a-zA-Z0-9-]*[a-zA-Z0-9])?$")
+
+// Host - holds network host IP/name and its port.
+type Host struct {
+	Name      string
+	Port      Port
+	IsPortSet bool
+}
+
+// IsEmpty - returns whether Host is empty or not
+func (host Host) IsEmpty() bool {
+	return host.Name == ""
+}
+
+// String - returns string representation of Host.
+func (host Host) String() string {
+	if !host.IsPortSet {
+		return host.Name
+	}
+
+	return host.Name + ":" + host.Port.String()
+}
+
+// Equal - checks whether given host is equal or not.
+func (host Host) Equal(compHost Host) bool {
+	return host.String() == compHost.String()
+}
+
+// MarshalJSON - converts Host into JSON data
+func (host Host) MarshalJSON() ([]byte, error) {
+	return json.Marshal(host.String())
+}
+
+// UnmarshalJSON - parses data into Host.
+func (host *Host) UnmarshalJSON(data []byte) (err error) {
+	var s string
+	if err = json.Unmarshal(data, &s); err != nil {
+		return err
+	}
+
+	// Allow empty string
+	if s == "" {
+		return nil
```

Comment:
In this case we need to set `host.Host = ""`, otherwise some stale value might remain right?

Reply:
Done